src/lib/operators/table_scan/column_vs_value_table_scan_impl.cpp

```diff
@@ -153,6 +153,11 @@ void ColumnVsValueTableScanImpl::_scan_sorted_segment(const BaseSegment& segment
                                                          boost::get<ColumnDataType>(value));
 
         sorted_segment_search.scan_sorted_segment([&](auto begin, auto end) {
+          if (begin == end) return;
+
+          // General note: If the predicate is NotEquals, there might be two matching ranges. scan_sorted_segment
+          // combines these two ranges into a single one. See sorted_segment_search.hpp for further details.
```

Comment:
Isn't this misleading? `scan_sorted_segment combines these two ranges into a single one` - the way I read this, the opposite happens. It does NOT combine them but calls the lambda twice.

Reply:
I think I misunderstood you in the first place. Let's clarify that offline.